var/spack/repos/builtin/packages/dpcpp/package.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+
+from spack import *
+
+
+class Dpcpp(CMakePackage):
+    """Data Parallel C++ compiler: Intel's implementation of SYCL programming model"""
+
+    homepage = 'https://intel.github.io/llvm-docs/'
+    git = 'https://github.com/intel/llvm.git'
+
+    version('develop', branch='sycl')
+    version('stable', commit='7981e6027a8cb43028be55d19f12cb1e30a28f5a')
+
+    maintainers = ['ravil-mobile']
+    variant('cuda', default=False, description='switch from OpenCL to CUDA')
+    variant('rocm', default=False, description='switch from OpenCL to ROCm')
+    variant('rocm-platform', default='AMD', values=('AMD', 'NVIDIA'), multi=False, description='choose ROCm backend')
+    variant('arm', default=False, description='build ARM support rather than x86')
+    variant('esimd-cpu', default=False, description='build with ESIMD_CPU support')
+    variant('assertions', default=False, description='build with assertions')
+    variant('docs', default=False, description='build Doxygen documentation')
+    variant('werror', default=False, description='Treat warnings as errors')
+    variant('shared', default=False, description='Build shared libraries')
+    variant('lld', default=False, description='Use LLD linker for build')
+
+    depends_on('cmake@3.16.2:', type='build')
+    depends_on('ninja@1.10.0:', type='build')
+
+    depends_on('cuda@10.2.0:10.2.999', when='+cuda')
+
+    # NOTE: AMD HIP needs to be tested; it will be done in the next update
+    # depends_on('cuda@10.2.0:10.2.999', when='rocm-platform=NVIDIA', type='build')
+    # depends_on('hip@4.0.0:', when='+rocm', type='build')
+
+    build_targets = ['deploy-sycl-toolchain']
+    install_targets = ['deploy-sycl-toolchain']
+
+    root_cmakelists_dir = 'llvm'
+
+    patch('cuda-backend.patch', when='@stable +cuda')
+
+    def cmake_args(self):
+        llvm_external_projects = 'sycl;llvm-spirv;opencl;libdevice;xpti;xptifw'
+
+        sycl_dir = os.path.join(self.stage.source_path, 'sycl')
+        spirv_dir = os.path.join(self.stage.source_path, 'llvm-spirv')
+        xpti_dir = os.path.join(self.stage.source_path, 'xpti')
+        xptifw_dir = os.path.join(self.stage.source_path, 'xptifw')
+        libdevice_dir = os.path.join(self.stage.source_path, 'libdevice')
+        llvm_targets_to_build = 'X86'
+        llvm_enable_projects = 'clang;' + llvm_external_projects
+        libclc_targets_to_build = ''
+        sycl_build_pi_rocm_platform = self.spec.variants['rocm-platform'].value
+
+        # replace not append, so ARM ^ X86
+        if '+arm' in self.spec:
+            llvm_targets_to_build = 'ARM;AArch64'
```

Comment:
The idea above is to avoid variants and use something like:
```suggestion
        if self.spec.satisfies('target=aarch64:'):
            llvm_targets_to_build = 'ARM;AArch64'
```

Reply:
Thanks for the idea. done